src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -123,9 +133,33 @@ CHIP_ERROR Spake2p::BeginVerifier(const uint8_t * my_identity, size_t my_identit
                                   size_t peer_identity_len, const uint8_t * w0in, size_t w0in_len, const uint8_t * Lin,
                                   size_t Lin_len)
 {
+#if ((CHIP_CRYPTO_HSM) && (ENABLE_HSM_SPAKE_VERIFIER))
+    uint8_t w0in_mod[32] = {
+        0,
+    };
+    size_t w0in_mod_len = 32;
```

Comment:
We have this in a bunch of places, and it looks like it can cause things to get out of sync, right?  Why not:
```
constexpr size_t w0in_mod_len = 32;
uint8_t w0in_mod[w0in_mod_len] = ....
```
And maybe even add template overloads of FELoad/FEWrite that will auto-deduce the length when a C array is used?  Followup for this, please?  @Jagadish-NXP

Reply:
regarding the ifdefs, point taken. We will be doing it the way pan-apple suggested below